### PR TITLE
[docs] Adapters

### PR DESCRIPTION
**Pull request by stevhliu**
Refactors the adapter docs.

### CODE REVIEW

**Review by sayakpaul — COMMENTED** [review]
Looking quite good. Wondering if merging of info from the different pages in the single page is okay for our users?

File: docs/source/en/tutorials/using_peft_for_inference.md [hunk 1]
```diff
@@ -10,218 +10,621 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+)
+# use cnmt in the prompt to trigger the LoRA
+pipeline("A cute cnmt eating a slice of pizza, stunning color scheme, masterpiece, illustration").images[0]
+```
+
+## torch.compile
 
-pipe_id = "stabilityai/stable-diffusion-xl-base-1.0"
-pipe = DiffusionPipeline.from_pretrained(pipe_id, torch_dtype=torch.float16).to("cuda")
+[torch.compile](../optimization/torch2.0#torchcompile) speeds up inference by compiling the PyTorch model to use optimized kernels. Before compiling, the LoRA weights need to be fused into the base model and unloaded first.
```
[the remark this answers is not in the document]
Comment (stevhliu) [reply docs/source/en/tutorials/using_peft_for_inference.md:106]: Good call, added a sentence about it with link!

**Comment by stevhliu** [comment]
Thanks Sayak! I think merging into a single page would make it easier for users to find everything they need instead of having users skip around several different docs to find it.

**Review by sayakpaul — COMMENTED** [review]
Significant work! I have left another round of comments. LMK if they make sense.

File: docs/source/en/tutorials/using_peft_for_inference.md [hunk 2]
```diff
@@ -10,218 +10,623 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+
+```py
+pipeline.unfuse_lora()
+```
+
+<div class="flex justify-center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/fuse_lora.png"/>
+</div>
+
+## Manage
```
[the remark this answers is not in the document]
Comment (stevhliu) [reply docs/source/en/tutorials/using_peft_for_inference.md:530]: Might be a little cleaner this way so users don't get all the extra stuff that comes with the docstrings and they can just focus on the code

File: docs/source/en/using-diffusers/ip_adapter.md [hunk 3]
```diff
@@ -197,60 +178,123 @@ scheduler = DDIMScheduler.from_pretrained(
...
+<hfoptions id="ipadapter-variants">
+<hfoption id="IP-Adapter Plus">
 
-IP-Adapter enabled pipelines provide the `ip_adapter_image_embeds` parameter to accept precomputed image embeddings. This is particularly useful in scenarios where you need to run the IP-Adapter pipeline multiple times because you have more than one image. For example, [multi IP-Adapter](#multi-ip-adapter) is a specific use case where you provide multiple styling images to generate a specific image in a specific style. Loading and encoding multiple images each time you use the pipeline would be inefficient. Instead, you can precompute and save the image embeddings to disk (which can save a lot of space if you're using high-quality images) and load them when you need them.
+```py
+import torch
+from transformers import CLIPVisionModelWithProjection, AutoPipelineForText2Image
 
-> [!TIP]
-> This parameter also gives you the flexibility to load embeddings from other sources. For example, ComfyUI image embeddings for IP-Adapters are compatible with Diffusers and should work ouf-of-the-box!
```
[the remark this answers is not in the document]
Comment (stevhliu) [reply docs/source/en/using-diffusers/ip_adapter.md:246]: This `[!TIP]` should still be here at line 311 :)

**Review by sayakpaul — APPROVED** [review]
